packages/opentelemetry-context-async-hooks/test/AsyncHooksContextManager.test.ts

```diff
@@ -166,6 +166,101 @@ describe('AsyncHooksContextManager', () => {
     });
   });
 
+  describe('.withAsync/with()', () => {
+    it('should correctly restore scope', async () => {
+      const scope1 = '1' as any;
+      const scope2 = '2' as any;
+      let done = false;
+
+      await contextManager.withAsync(scope1, async () => {
+        assert.strictEqual(contextManager.active(), scope1);
+        contextManager.with(scope2, () => {
+          assert.strictEqual(contextManager.active(), scope2);
+          done = true;
+        });
+        assert.strictEqual(contextManager.active(), scope1);
+      });
+
+      assert.ok(done);
+    });
+
+    it('should correctly restore scope', done => {
+      const scope1 = '1' as any;
+      const scope2 = '2' as any;
+
+      contextManager.with(scope1, async () => {
+        assert.strictEqual(contextManager.active(), scope1);
+        await contextManager.withAsync(scope2, async () => {
+          assert.strictEqual(contextManager.active(), scope2);
+          return done();
```

Comment:
Isn't `done()` called here in the sync part of the async function? e.g. before the `        assert.strictEqual(contextManager.active(), scope1);` is reached?